core-processor/src/ext.rs

```diff
@@ -187,12 +187,17 @@ impl ProcessorExt for Ext {
 }
 
 impl IntoExtInfo for Ext {
-    fn into_ext_info<F: FnMut(usize, &mut [u8])>(self, mut get_page_data: F) -> ExtInfo {
+    fn into_ext_info<F: FnMut(usize, &mut [u8]) -> Result<(), &'static str>>(
+        self,
+        mut get_page_data: F,
+    ) -> Result<ExtInfo, (&'static str, GasAmount)> {
         let accessed_pages_numbers = self.allocations_context.allocations().clone();
         let mut accessed_pages = BTreeMap::new();
         for page in accessed_pages_numbers {
             let mut buf = alloc::vec![0u8; PageNumber::size()];
-            get_page_data(page.offset(), &mut buf);
+            if let Err(err) = get_page_data(page.offset(), &mut buf) {
```

Comment:
Can't we use map_err and ? syntax

Reply:
We cant because closure will move self. Do you have any ideas how to operate better with this case?